plugin/src/test/java/io/jenkins/plugins/casc/misc/Util.java

```diff
@@ -229,34 +234,47 @@ public static Schema returnSchema() throws Exception{
     }
 
     /**
-     * Validates a given jsonObject against the schema generated for the current live jenkins instance
-     *      * Example Usage:
-     *      *      * <pre>{@code
-     *      *      * assertTrue(validateSchema(jsonSubject));}
-     *      *      * </pre>
-     * @param  jsonSubject The json Object that needs to be validated
-     * @return true if it's valid else returns false
+     * Validates a given jsonObject against the schema generated for the current live jenkins
+     * instance.
+     *
+     * <p>Example Usage:</p>
+     *  <pre>{@code
+     *   assertThat(validateSchema(convertYamlFileToJson(this, "invalidSchemaConfig.yml")),
+     *             contains("#/jenkins/numExecutors: expected type: Number, found: String"));
+     *  }</pre
+     *
+     *  <pre>{@code
+     *   assertThat(validateSchema(convertYamlFileToJson(this, "validConfig.yml")),
+     *             empty());
+     *  }</pre>
+     *
+     * @param jsonSubject the json object that needs to be validated
+     * @return a list of validation errors, empty list if no errors
      */
-    public static boolean validateSchema(JSONObject jsonSubject) {
+    public static List<String> validateSchema(JSONObject jsonSubject) {
         try {
             returnSchema().validate(jsonSubject);
+        } catch (ValidationException e) {
+            return e.getAllMessages();
         } catch (Exception ie) {
```

Comment:
Is this still required, would reducing them to one block make sense

Reply:
What do you mean? this is the whole point of the change, the validation messages are now returned as a list.